pallets/gear/src/lib.rs

```diff
@@ -1193,6 +1194,8 @@ pub mod pallet {
             gas_limit: u64,
             value: BalanceOf<T>,
         ) -> Result<InitPacket, DispatchError> {
+            let program_id = ProgramId::generate(code_id, &salt);
```

Comment:
what's this? it's a wrong program id to be checked.

Reply:
there are no program ids that are generated from salt and code id only.

Reply:
Not sure how to replace this, might need to remove the check from here and put it elsewhere